### PR TITLE
V2 sample file updates and tweaks

### PR DESCRIPTION
**Pull request by guymaness2**
##### What does the change add or fix?
  
* Update csv files for Bulldozer v2.
* Minor tweaks required while testing v2 sample csv import.

---------

##### What does the change add or fix in a succinct statement that will be read by clients?

* Update csv files for Bulldozer v2.
* Minor tweaks required while testing v2 sample csv import.

---------

##### Who reported, requested, or paid for the change?

Warranty

---------

##### Does this update or add options to the block UI?

none

---------

##### What files does it affect?

[Add a file list here]

---------

##### Is it a breaking change for other versions/clients?

no

### CODE REVIEW

File: Bulldozer.CSV/Maps/Attributes.cs [hunk 1]
```diff
@@ -84,20 +84,12 @@ private int LoadEntityAttributes( CSVInstance csvData )
...
-                    var fk = string.Empty;
-                    if ( string.IsNullOrWhiteSpace( attributeForeignKey ) )
-                    {
-                        fk = string.Format( "Bulldozer_{0}_{1}", categoryName.RemoveWhitespace(), attributeName.RemoveWhitespace() ).Left( 100 );
-                    }
-                    else
-                    {
-                        fk = attributeForeignKey;
-                    }
+                    var attributeForeignKey = $"{this.ImportInstanceFKPrefix}^{rockKey}_{entityTypeNameArray.LastOrDefault()}";
```
Comment (guymaness2) [inline Bulldozer.CSV/Maps/Attributes.cs:87]:
```suggestion
                    var attributeForeignKey = $"{this.ImportInstanceFKPrefix}^{rockKey}_{entityTypeNameArray.LastOrDefault()}".Left( 100 );
```

File: Bulldozer.CSV/Maps/ConnectionRequest.cs [hunk 2]
```diff
@@ -126,7 +126,22 @@ private int LoadConnectionRequest( CSVInstance csvData )
 
                     if ( request == null && requestor != null && requestStatus != null )
                     {
-                        request = AddConnectionRequest( opportunity, rForeignKey, rCreatedDate, rModifiedDate, requestStatus.Id, ( ConnectionState ) rState, rComments, rFollowUp, requestor.PersonAliasId, requestConnector?.PersonAliasId );
+                        request = new ConnectionRequest
```
Comment (guymaness2) [inline Bulldozer.CSV/Maps/ConnectionRequest.cs:129]: remove original method now that it is not being used.

File: Bulldozer.CSV/Maps/Financial.cs [hunk 3]
```diff
@@ -921,14 +936,34 @@ private int LoadScheduledTransaction( CSVInstance csvData )
...
+                    if ( rowAccountId.HasValue )
+                    {
+                        accountFk += $"{rowAccountId.Value}";
+                    }
+                    else
+                    {
+                        accountFk += rowAccount.RemoveWhitespace().Truncate( 50 );
+                    }
+                    account = accountList.FirstOrDefault( a => a.ForeignKey == accountFk );
+                    if ( account == null )
```
Comment (guymaness2) [inline Bulldozer.CSV/Maps/Financial.cs:956]:
```suggestion
                    if ( account == null && !rowAccountId.HasValue )
```

File: Bulldozer.CSV/Maps/Group.cs [hunk 4]
```diff
@@ -777,6 +789,17 @@ private int ImportGroupAddresses()
                     ReportPartialProgress();
                 }
             }
+            if ( groupTypesToUpdate.Count > 0 )
+            {
+                groupTypesToUpdate = groupTypesToUpdate.Distinct().ToList();
+                var groupTypes = new GroupTypeService( rockContext ).Queryable().Where( gt => groupTypesToUpdate.Contains( gt.Id ) );
+                foreach ( var groupType in groupTypes )
+                {
+                    groupType.LocationSelectionMode = GroupLocationPickerMode.Address;
```
Comment (guymaness2) [inline Bulldozer.CSV/Maps/Group.cs:798]: Look into adding 1 to value vs. direct set.

File: Bulldozer.CSV/Maps/Person.cs [hunk 5]
```diff
@@ -63,7 +63,7 @@ private int ImportPersonList()
 
                                                 } );
 
-            int nextNewFamilyForeignId = this.FamilyDict.Any( a => a.Value.ForeignId.HasValue ) ? this.FamilyDict.Max( a => a.Value.ForeignId.Value ) : 0;
+            int nextNewFamilyForeignId = this.FamilyDict.Any( a => a.Value.ForeignId.HasValue ) ? this.FamilyDict.Max( a => a.Value.ForeignId.GetValueOrDefault() ) : 0;
```
Comment (guymaness2) [inline Bulldozer.CSV/Maps/Person.cs:66]: What is default?

File: Bulldozer.CSV/SampleCSVs/Legacy/01 - Individual.csv
Comment (nateh777) [inline Bulldozer.CSV/SampleCSVs/Legacy/01 - Individual.csv:1]: Remove Legacy folder, not necessary.

File: Bulldozer.CSV/Maps/ConnectionRequest.cs (hunk 2 above)
Comment (guymaness2) [reply Bulldozer.CSV/Maps/ConnectionRequest.cs:129]: Had to update F1 notes map to same structure to be able to get rid of the method

File: Bulldozer.CSV/Maps/Person.cs (hunk 5 above)
Comment (guymaness2) [reply Bulldozer.CSV/Maps/Person.cs:66]: Per debugging, default pulls 0 for null values.

**Comment by guymaness2** [comment]
@nateh777 all conversations resolved.

**Review by nateh777 — APPROVED** [review]

**Pull request merged**